src/providers/azure-ad-b2c.js

```diff
@@ -32,10 +36,24 @@ export default function AzureADB2C(options) {
         name = `${profile.given_name}`
       }
 
+      if (profile.email) {
+        email = profile.email
+      }
+      else if (profile.emails) {
+        email = profile.emails[0]
+      }
+
+      if (profile.oid) {
+        id = profile.oid
+      }
+      else if (profile.sub) {
+        id = profile.sub
+      }
+
       return {
-        id: profile.oid,
+        id,
         name,
-        email: profile.emails[0],
+        email,
         image: null,
       }
```

Comment:
The way the properties are returned with a Custom Policy is rather different than the way the properties were returned with an out-of-the-box User Flow.

Reply:
We likely want to go with the most basic flow for the built-in provider though, and rather educate people better on how to achieve custom behaviour in https://next-auth.js.org/providers/azure-ad-b2c